usvm-ts/src/main/kotlin/org/usvm/machine/operator/TsBinaryOperator.kt

```diff
@@ -97,294 +99,274 @@ sealed interface TsBinaryOperator {
         return internalResolve(lhsValue, rhsValue, scope)
     }
 
-    data object Eq : TsBinaryOperator {
-        override fun TsContext.onBool(
-            lhs: UBoolExpr,
-            rhs: UBoolExpr,
-            scope: TsStepScope,
-        ): UBoolExpr {
-            return mkEq(lhs, rhs)
-        }
+    fun <R : USort> TsBinaryOperator.commonResolveFakeObject(
```

Comment:
This function does not need to be an extension of `TsBinaryOperator`, since this is already a method of this class. Weird how IDEA does not warn about this meaningless construction.

You probably want to make it an extension of `TsContext`, similar to `onXXX` methods defined above. That way, you could eliminate nasty `with (lhs.tctx)` block.